.github/workflows/build.yml

```
@@ -14,49 +28,107 @@
+      
       - name: Add execute permissions
+        if: steps.should-build.outputs.should_run == 'true'
         run: chmod +x bin/*
         shell: bash
 
       # Compress build artifact
       - name: Compress build artifact
-        run: cd ./bin; zip -j ../${{ matrix.build.zip-name }}.zip arco
+        if: steps.should-build.outputs.should_run == 'true'
+        id: compress
+        run: |
```

actionlint (1.7.7)

shellcheck reported issue in this script: SC2086:info:4:30: Double quote to prevent globbing and word splitting

(shellcheck)